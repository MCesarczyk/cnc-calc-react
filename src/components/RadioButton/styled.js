import styled from "styled-components";

export const RadioButtonWrapper = styled.div`
    display: flex;
    gap: 10px;
`;