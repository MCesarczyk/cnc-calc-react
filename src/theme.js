export const theme = {
    color: {
        primary: "#1e3246",
        primaryContrast: "#fff",
        background: "#133246",
        submitButtonFont: "#19aa6e",
        resetButtonFont: "#dc3223",
        invalid: "beige",
        disabled: "#ddd"
    },
    breakpoint: {
        xs: "480px",
        md: "768px"
    },
}