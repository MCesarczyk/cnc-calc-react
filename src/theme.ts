export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
