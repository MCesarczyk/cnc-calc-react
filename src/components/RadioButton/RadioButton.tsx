import { ChangeEvent } from 'react';
import styled from 'styled-components';

import { Input } from '../Input';
import { TextSub } from '../TextSub/TextSub';
import { FeedOption } from 'types';

interface RadioButtonProps {
  name: string;
  option: FeedOption;
  parameter: string;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

export const RadioButton = ({ name, option, parameter, onChange }: RadioButtonProps) => {
  return (
    <RadioButtonWrapper>
      <Input
        type="radio"
        name={name}
        id={option?.id}
        data-testid={`${option?.id}-radio`}
        value={option?.id}
        checked={parameter === option?.id}
        onChange={onChange}
      />
      <label htmlFor={option?.id}>
        {option?.name}
        <TextSub>{option?.sub}</TextSub>
        {option?.unit}
      </label>
    </RadioButtonWrapper>
  );
};

const RadioButtonWrapper = styled.div`
  display: flex;
  gap: 10px;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    align-self: flex-start;
  }
`;
