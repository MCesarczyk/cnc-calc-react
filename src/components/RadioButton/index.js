import { TextSub } from "../TextSub";
import { RadioButtonWrapper } from "./styled";

const RadioButton = ({ name, option, parameter, onChange }) => (
  <RadioButtonWrapper>
    <input
      type="radio"
      name={name}
      id={option?.id}
      data-testid={`${option?.id}-radio`}
      value={option?.id}
      checked={parameter === option?.id}
      onChange={onChange}
    />
    <label htmlFor={option?.id}>
      {option?.name}
      <TextSub>
        {option?.sub}
      </TextSub>
      {option?.unit}
    </label>
  </RadioButtonWrapper>
);

export default RadioButton;