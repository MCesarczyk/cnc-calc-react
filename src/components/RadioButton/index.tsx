import { ChangeEvent } from "react";

import { Input } from "../Input";
import { TextSub } from "../TextSub";
import { RadioButtonWrapper } from "./styled";

interface RadioButtonProps {
  name: string;
  option: {
    id: number;
    name: string;
    sub: string;
    unit: string;
  };
  parameter: number;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

const RadioButton = ({
  name,
  option,
  parameter,
  onChange,
}: RadioButtonProps) => {
  return (
    <RadioButtonWrapper>
      <Input
        type="radio"
        name={name}
        id={option?.name}
        data-testid={`${option?.id}-radio`}
        value={option?.id}
        checked={parameter === option?.id}
        onChange={onChange}
      />
      <label htmlFor={option?.name}>
        {option?.name}
        <TextSub>{option?.sub}</TextSub>
        {option?.unit}
      </label>
    </RadioButtonWrapper>
  );
};

export default RadioButton;
