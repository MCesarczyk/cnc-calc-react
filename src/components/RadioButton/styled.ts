import styled from 'styled-components';

export const RadioButtonWrapper = styled.div`
  display: flex;
  gap: 10px;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    align-self: flex-start;
  }
`;
