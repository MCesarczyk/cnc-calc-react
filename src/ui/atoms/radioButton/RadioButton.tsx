import { ChangeEvent, CSSProperties } from 'react';
import styled from 'styled-components';

import { FeedOption } from 'types';
import { TextSub } from 'ui/atoms/textSub';
import { Input } from 'ui/atoms/input';

export type RadioButtonVariants = 'DEFAULT';

type RadioButtonStyles = Required<
  Pick<CSSProperties, 'color' | 'backgroundColor' | 'borderColor'> & {
    disabledText?: CSSProperties['color'];
    disabledBackgroundColor?: CSSProperties['backgroundColor'];
  }
>;

export type Palette<TVariants extends string, TStyles> = {
  [variant in TVariants]: TStyles;
};

const styles: Palette<RadioButtonVariants, RadioButtonStyles> = {
  DEFAULT: {
    color: 'primaryText',
    backgroundColor: 'elementBackground',
    borderColor: 'primary',
    disabledText: 'disabledText',
    disabledBackgroundColor: 'disabledBackground',
  },
};

interface RadioButtonProps {
  name: string;
  option: FeedOption;
  parameter: string;
  disabled?: boolean;
  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
}

export const RadioButton = ({ name, option, parameter, disabled, onChange }: RadioButtonProps) => {
  return (
    <RadioButtonWrapper>
      <Input
        type="radio"
        name={name}
        id={option?.id}
        data-testid={`${option?.id}-radio`}
        value={option?.id}
        checked={parameter === option?.id}
        disabled={disabled}
        onChange={onChange}
      />
      <label htmlFor={option?.id}>
        {option?.name}
        <TextSub>{option?.sub}</TextSub>
        {option?.unit}
      </label>
    </RadioButtonWrapper>
  );
};

const RadioButtonWrapper = styled.div`
  display: flex;
  gap: 10px;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    align-self: flex-start;
  }
`;

RadioButton.defaultProps = {};

RadioButton.displayName = 'RadioButton';
