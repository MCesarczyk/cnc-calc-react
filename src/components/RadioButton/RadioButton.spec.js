import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { theme } from 'theme';
import { RadioButton } from './RadioButton';

test('Radio button rendered properly', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <RadioButton />
    </ThemeProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
