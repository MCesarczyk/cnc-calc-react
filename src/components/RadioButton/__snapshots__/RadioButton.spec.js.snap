// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio button rendered properly 1`] = `
.c1 {
  vertical-align: sub;
  font-size: 0.7em;
  margin-right: 0.15em;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
}

@media (max-width:480px) {
  .c0 {
    -webkit-align-self: flex-start;
    -ms-flex-item-align: start;
    align-self: flex-start;
  }
}

<div
  className="c0"
>
  <input
    checked={true}
    type="radio"
  />
  <label>
    <span
      className="c1"
    />
  </label>
</div>
`;
